src/anyp/Host.h

```diff
@@ -0,0 +1,81 @@
+/*
+ * Copyright (C) 1996-2023 The Squid Software Foundation and contributors
+ *
+ * Squid software is distributed under GPLv2+ license and includes
+ * contributions from numerous individuals and organizations.
+ * Please see the COPYING and CONTRIBUTORS files for details.
+ */
+
+#ifndef SQUID_SRC_ANYP_HOST_H
+#define SQUID_SRC_ANYP_HOST_H
+
+#include "dns/forward.h"
+#include "ip/Address.h"
+#include "sbuf/SBuf.h"
+
+#include <iosfwd>
+#include <optional>
+#include <variant>
+
+namespace AnyP
+{
+
+/// either a domain name (as defined in DNS RFC 1034) or an IP address
+class Host
```

Comment:
This class is at the core of this PR changes. In official code, a single address that may be either an IP address or a DNS domain name is either isolated into two or more co-existing objects (e.g., host information inside AnyP::Uri) or morphed into one opaque SBuf object (e.g., `server` in ssl_verify_cb()). Neither approach works well, for obvious reasons. This PR uses the new Host class to accurately and as-safely-as-possible represent this naming duality in several X.509 SAN-related contexts. Eventually, more code should be converted to use this class. I have left a few related TODOs.

The proposed "host" name works well enough IMO, but I would not be surprised if there are better alternatives our there.

Reply:
Hi @rousskov - do you know if there's anything else that needs to be done for the review to continue? Thanks.

Reply:
@walkert, unfortunately, the answer to your question is "no": I do not know how to convince @yadij to review PRs awaiting his review. IMO, this persistent problem is one of the existential threats faced by the Squid Project, but all my attempts to find a solution have failed. I can only hope that we will resume making progress with this PR soon.

Reply:
@rousskov understood. Thanks for taking the time to respond. I'll keep watching and hoping.

Reply:
Hi @rousskov, is there scope to hand this off to another reviewer if @yadij is unavailable? Thanks.

Reply:
@walkert, there is no formal procedure to overwrite, move, or clear a "this PR needs to be reviewed by me" flag set by a core developer. Project core developers hold a lot of power, and multiple attempts to establish timeouts (or otherwise restructure the governing rules) have failed so far. FWIW, I plan to bring this PR up again during the next meeting which should happen on October 22, 2024.

Reply:
@rousskov understood. Thanks for the info. 🙏